### ISSUE TITLE
Support ES Module

### ISSUE THREAD

**Issue by marchbox** [blinkk/amagaki#53]
I'm exploring the option of using amagaki within an existing project that uses ES Module. Here is an illustration of what the directory structure looks like:

```
project root
├ package.json (with `"type": "module"`)
├ …
└ website
     ├ package.json (WITHOUT `"type": "module"`)
     ├ amagaki.js
     └  …
```

The reason the amagaki project is under another project is because the parent project contains both dynamic pages (server-rendered) and static marketing pages, and I wanted to use amagaki as the static site generator for the marketing pages for easy maintenance.

It's currently working but would be nice to have everything under the “website” directory using the same module system.

Let me know if I'm missing something.

**Comment by jeremydw** [comment]
I don't think this should be a problem. The `amagaki-starter` project does assume one TS/Sass stack but Amagaki core can be used within any environment as the static generator.

If `"type": "module"` is used, can you elaborate on which bit isn't working?

**Comment by marchbox** [comment]
Yeah, right now if I don't place the extra package.json to reset the type, when running `npx amagaki serve storefront` (storefront is the subdirectory that contains the code for marketing pages), it'll throw this:

```
/Users/zacky/Code/epos-storefront/node_modules/@amagaki/amagaki/src/index.js:32
    throw up;
    ^

Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/zacky/Code/epos-storefront/storefront/amagaki.js
require() of ES modules is not supported.
require() of /Users/zacky/Code/epos-storefront/storefront/amagaki.js from /Users/zacky/Code/epos-storefront/node_modules/@amagaki/amagaki/src/pod.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename amagaki.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/zacky/Code/epos-storefront/package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at new Pod (/Users/zacky/Code/epos-storefront/node_modules/@amagaki/amagaki/src/pod.js:80:39)
    at ServeCommand.run (/Users/zacky/Code/epos-storefront/node_modules/@amagaki/amagaki/src/commands/serve.js:34:21)
    at Command.<anonymous> (/Users/zacky/Code/epos-storefront/node_modules/@amagaki/amagaki/src/index.js:55:9)
    at Command.listener [as _actionHandler] (/Users/zacky/Code/epos-storefront/node_modules/@amagaki/amagaki/node_modules/commander/index.js:922:31)
    at Command._parseCommand (/Users/zacky/Code/epos-storefront/node_modules/@amagaki/amagaki/node_modules/commander/index.js:1503:14) {
  code: 'ERR_REQUIRE_ESM'
}
```

My amagaki.js looks like this:

```js
export default function(pod) {
  pod.configure({
    metadata: {
      name: 'Amagaki Starter',
    },
    localization: {
      defaultLocale: 'en',
      locales: ['en'],
    },
    staticRoutes: [],
  });
}
```

**Comment by marchbox** [comment]
Chatted offline, node v10 doesn't require the extra package.json for amagaki to work, but node v14 does require it.

**Comment by jeremydw** [comment]
We've made some improvements, namely supporting auto-transpilation of `.ts` files without `tsc` as well as some updates to Amagaki's package file that might improve support here.

I added a test for your exact setup, and tested it on Node 14:
https://github.com/blinkk/amagaki/tree/main/tests/modules

I think this should be working now, with the latest version of Amagaki as well as the usage of the `amagaki.ts` file. Let me know if you're still having issues.
